Rename `head` to `tracking_ref` in `GitDagBundle`

dstandish opened the pull request:
Head is confusing cus in git speak it means "what you have checked out".  That's not what we mean here.  Here we're trying to describe the branch in the repo that the user wants this bundle to track or "follow".  Technically it could be a tag but that makes less sense since you shouldn't really be moving tags should you...

jedcunningham commented on airflow/dag_processing/bundles/git.py:
```diff
@@ -38,15 +38,15 @@ class GitDagBundle(BaseDagBundle):
 
     :param repo_url: URL of the git repository
-    :param head: Branch or tag for this DAG bundle
+    :param branch: Branch or tag for this DAG bundle
```
Let's do `ref` instead - that's a better name since it can be either a branch or a tag (not a moving tag, but the bundle can be reconfigured to point at a new tag!).

dstandish replied:
i did `tracking_ref`

dstandish committed:
Rename head to tracking ref

Head is confusing cus in git speak it means "what you have checked out".  That's not what we mean here.  Here we're trying to describe, most commonly, the branch in the repo that the user wants this bundle to track or "follow".  Technically it could also be a tag.
